A few smaller fixes and a test output improvement

dean0x7d opened the pull request:
The fixes are described in the first two commit messages.

The skipped test bug silently passes when it shouldn't. It came up as an error while I was working on another feature (I'll submit a separate PR later), but it gave me a bit of trouble. That was the motivation for the third commit: diff output for failed tests. For example, with the change a failure report would look like this:
```
Test "example2" FAILED!
--- output
+++ reference
  ('test', True)
  (5, 'test', True)
  15
  5
  Destructing Example2
  Entry at positon 0: value
  Example2: No constructor defined!
  Help on class Example2 in module example
  This exception was intentionally thrown.
  ['array entry 1', 'array entry 2']
  __module__(Example2) = example
  __module__(Example2.get_set) = example
  __name__(Example2) = Example2
  __name__(Example2.get_set) = get_set
  __name__(example) = example
  array item 0: array entry 1
  array item 1: array entry 2
  can't set attribute
  class EExxaammppllee22(builtins.object)
  key: key, value=value
- key: key, value=value_bug
?                      ----
+ key: key, value=value
  key: key1
  key: key1
  key: key2
```
This is a [Differ-style delta from difflib](https://docs.python.org/3.5/library/difflib.html#difflib.ndiff).

The last commit adds a `make check` target to CMake which will rebuild as needed and test, unlike `make test` which only tests. This is more useful when there are multiple targets in a project, but the tests only rely on a subset. This is not the case for pybind11, but it's still nice to have for two reasons: 1) With `make test` it's possible to forget to rebuild first, with `make check` it's not. 2) For integration with some IDEs.

wjakob commented:
Looks great -- do you also want to enable the 'check' target on Travis/AppVeyor? (see the .yml files in the root directory)

dean0x7d commented:
I updated the CI configurations. While I was in there I also changed the numpy install to the new [manylinux](https://github.com/pypa/manylinux) binary wheel. It makes the installation practically instant, reducing the Travis build by about 2 minutes.

wjakob commented:
Awesome! Thank you, that's much appreciated!
